### DISCUSSION TITLE
Presentation content from multiple files

### DISCUSSION THREAD

**Discussion by sradhakrishna** [marp-team/marp#180] [Ideas] [answered]
A few ideas to consider:

1. Allow a slide deck to be generated from a folder structure, each folder turning to a section within the presentation. (Usecase: A master slide deck for a course, where different sections are taught by different individuals
    a. Allows different sections of a presentation to be organized in different folders
    b. Embedded content - such as images etc can be even organized in a folder
    c. Such a folder structure, in a git repository can enable code reviews of targeted sections of the presentation
2. Like Github pages, there could be Github slides - slides.github.com - as well. :-)

**Comment by yhatt** [comment, marked as answer]
A basic principle of Marp ecosystem is "a single Markdown file corresponds to a single presentation deck" so Marp team would not work for getting the slide consisted by multiple files.

I've already described our stances for this principle in https://github.com/marp-team/marpit/issues/257#issuecomment-695783863.

Of course, we also do never disturb community making [a Marp plugin](https://marpit.marp.app/usage?id=extend-marpit-by-plugins) or glue tool for concating contents! For example, [a compatible markdown-it-include plugin](https://www.npmjs.com/package/markdown-it-include) can import the Markdown content from other fragment file(s). [Here is the use case in the wild.](https://github.com/sergio-fry/slides/tree/master/modular-app)

This idea should be helpful to make a glue tool to merge multiple Markdown contents in sub directories into one Markdown for Marp conversion.

### Related issues

- https://github.com/marp-team/marpit/issues/227
- https://github.com/marp-team/marp-core/issues/163
- https://github.com/marp-team/marpit/issues/257

**Comment by yhatt** [comment]
> 2. Like Github pages, there could be Github slides - slides.github.com - as well. :-)

GitHub does not have like a GitHub slides, but you can preview Marp slides without leaving GitHub, powered by [github.dev](https://github.dev) and [Marp for VS Code](https://github.com/marp-team/marp-vscode)! After opened Marp Markdown document in GitHub, just hit <kbd>.</kbd> key! You can preview and edit slides on the web by installing Marp extension.

And there are some pre-built workflows for GitHub Actions to publish Marp slides into GitHub pages: 

- [marp-action](https://github.com/marketplace/actions/marp-action)
- [Marp CLI Action](https://github.com/marketplace/actions/marp-cli-action)
